### DISCUSSION TITLE
How to ignore folders e.g. git and vscode folders

### DISCUSSION THREAD

**Discussion by AceTheWiz** [statiqdev/Discussions#8] [Q&A]
Maybe I'm missing something but how do I instruct Statiq.Web to ignore .git and .vscode folders.

**Comment by daveaglick** [comment]
Like most things in Statiq, you've got a couple options off the top of my head:

- The pipelines are controlled by settings that control where they look for files (`ContentFiles`, `DataFiles`, and `AssetFiles` in the next release). You can set new [globbing patterns](https://statiq.dev/framework/execution/files#globbing) to change exactly where and while files are found.
- [You can add](https://statiq.dev/web/content-and-data/metadata/directory-metadata) a `_directory.yml` file to any directory you want to be totally [excluded](https://statiq.dev/web/content-and-data/excluding-documents) and add `Excluded: true` to it.

Can I ask how you ended up with `.git` and `.vscode` folders being processed in the first place? Statiq only looks in the `input` folder unless you tell it otherwise, specifically to avoid getting the files that should be processed confused with things like the .NET project, git repository, etc. Do you have `.git` and `.vscode` folders under your `input` folder?

**Reply by AceTheWiz** [reply]
It was in the Asset pipeline.
I did look for an AssetFiles configuration setting but it was not there at the time (I will probably use this when it is available).
My content (aka Input) is in a separate repository. So yes it has ```.git``` and ```.vscode``` folders. Other people generate and maintain the website content so I do not want them to see the generator files. Make sense. ?
BTW Is there a nuget source for the bleeding Statiq releases.

**Reply by daveaglick** [reply]
Ah, gotcha - that's an interesting use case, I definitely want to make sure it's supported. The new `AssetFiles` setting is out on NuGet in Statiq.Web 1.0.0-beta.2 so you can try that out.

I wonder if it also makes sense to add a setting for folder names that should be completely excluded no matter where they are or at what depth and default it to some known ones like `.git`, `.vs`, `.vscode`, etc.

**Reply by AceTheWiz** [reply]
Thanks Dave.

I will try the ```AssetFiles``` setting.
Sorry to bother you with this question but I cannot get my head around the globbing syntax. I have searched the net but cannot find an authorative source. 

What is the pattern to exclude folders starting with a '.' or '_'? 

I tried this but I can't tell if its working. (because of multiple other issues I'm having).
```C#
.AddSetting(WebKeys.AssetFiles, "{!.,!_,}**/*")
```

As for excluding entire folders, it would suit my use case of course 😃 but is it good for all 🤔
